tests/integration/security/authorization_test.go

```diff
@@ -119,78 +115,80 @@ func TestAuthorization_JWT(t *testing.T) {
-				With(&d, util.EchoConfig("d", ns, false, nil, nil)).
-				BuildOrFail(t)
-
-			newTestCase := func(target echo.Instance, namePrefix string, jwt string, path string, expectAllowed bool) rbacUtil.TestCase {
-				return rbacUtil.TestCase{
-					NamePrefix: namePrefix,
-					Request: connection.Checker{
-						From: a,
-						Options: echo.CallOptions{
-							Target:   target,
-							PortName: "http",
-							Scheme:   scheme.HTTP,
-							Path:     path,
-						},
-					},
-					Jwt:           jwt,
-					ExpectAllowed: expectAllowed,
-				}
-			}
-			cases := []rbacUtil.TestCase{
-				newTestCase(b, "[NoJWT]", "", "/token1", false),
-				newTestCase(b, "[NoJWT]", "", "/token2", false),
-				newTestCase(b, "[Token1]", jwt.TokenIssuer1, "/token1", true),
-				newTestCase(b, "[Token1]", jwt.TokenIssuer1, "/token2", false),
-				newTestCase(b, "[Token2]", jwt.TokenIssuer2, "/token1", false),
-				newTestCase(b, "[Token2]", jwt.TokenIssuer2, "/token2", true),
-				newTestCase(b, "[Token1]", jwt.TokenIssuer1, "/tokenAny", true),
-				newTestCase(b, "[Token2]", jwt.TokenIssuer2, "/tokenAny", true),
-				newTestCase(b, "[PermissionToken1]", jwt.TokenIssuer1, "/permission", false),
-				newTestCase(b, "[PermissionToken2]", jwt.TokenIssuer2, "/permission", false),
-				newTestCase(b, "[PermissionTokenWithSpaceDelimitedScope]", jwt.TokenIssuer2WithSpaceDelimitedScope, "/permission", true),
-				newTestCase(b, "[NoJWT]", "", "/tokenAny", false),
-				newTestCase(c, "[NoJWT]", "", "/somePath", true),
-
-				// Test condition "request.auth.principal" on path "/valid-jwt".
-				newTestCase(d, "[NoJWT]", "", "/valid-jwt", false),
-				newTestCase(d, "[Token1]", jwt.TokenIssuer1, "/valid-jwt", true),
-				newTestCase(d, "[Token1WithAzp]", jwt.TokenIssuer1WithAzp, "/valid-jwt", true),
-				newTestCase(d, "[Token1WithAud]", jwt.TokenIssuer1WithAud, "/valid-jwt", true),
-
-				// Test condition "request.auth.presenter" on suffix "/presenter".
-				newTestCase(d, "[Token1]", jwt.TokenIssuer1, "/request/presenter", false),
-				newTestCase(d, "[Token1WithAud]", jwt.TokenIssuer1, "/request/presenter", false),
-				newTestCase(d, "[Token1WithAzp]", jwt.TokenIssuer1WithAzp, "/request/presenter-x", false),
-				newTestCase(d, "[Token1WithAzp]", jwt.TokenIssuer1WithAzp, "/request/presenter", true),
-
-				// Test condition "request.auth.audiences" on suffix "/audiences".
-				newTestCase(d, "[Token1]", jwt.TokenIssuer1, "/request/audiences", false),
-				newTestCase(d, "[Token1WithAzp]", jwt.TokenIssuer1WithAzp, "/request/audiences", false),
-				newTestCase(d, "[Token1WithAud]", jwt.TokenIssuer1WithAud, "/request/audiences-x", false),
-				newTestCase(d, "[Token1WithAud]", jwt.TokenIssuer1WithAud, "/request/audiences", true),
+			for _, srcCluster := range ctx.Clusters() {
+				ctx.NewSubTest(fmt.Sprintf("From a In %s", srcCluster.Name())).Run(func(ctx framework.TestContext) {
+					a := apps.A.Match(echo.InCluster(srcCluster).And(echo.NameSpace(ns.Name())))
```

Comment:
A general question as I'm not following the multi cluster test setup, do we deploy the application (a, b, c, d) in multiple clusters by default? And the for-loop here tests each cluster separately (e.g. a in cluser1, cluster2, ...)? Does this involve any cross-cluster traffic? thanks.

Reply:
Answer to 
1) Yes, when we deploy a, b, c, d, we do the loop on all clusters https://github.com/istio/istio/pull/28088/files/a9c1913a105b0308e10358baaae01723f9426eb0#diff-84045abbcab1322dfe6752d85763aee916a8811a98783438cdae5d0e4a05aac7R140-R143. 
2) for-loop will loop src clusters separately, so we init the call for every service within every clusters.
3) After the call is made from the src, destinations will be load balanced among clusters within the same target service. That is when cross-cluster traffic involves and we validation we are able to reach all clusters via multiple calls.